test/internal/helpers/CompareContractValidator.java

```diff
@@ -0,0 +1,110 @@
+package internal.helpers;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CompareContractValidator {
+  // Helper method to normalize comparisons to [-1, 0, 1] before testing
+  private static int sgn(int value) {
+    return Integer.compare(value, 0);
+  }
+
+  public static class Violator {
+    final String first;
+    final String second;
+    final String reason;
+
+    public Violator(String a, String b, String c) {
+      first = a;
+      second = b;
+      reason = c;
+    }
+  }
+
+  public enum ViolationType {
+    SYMMETRY,
+    FUNCTION_EQUALITY,
+    OPERATOR_EQUALITY,
+    TRANSITIVITY
+  }
+
+  // Code intended to verify that compareTo() meets its contract.
+  // public static <T extends Comparable<T>> List<T> checkForContractViolations(List<T> inputList) {
+  public static <T> List<Violator> checkForContractViolations(List<T> inputList) {
+
+    List<Violator> violators = new ArrayList<>();
```

Comment:
This list and the associated structures were the first pass at capturing violations in the context of a unit test.  Not sure it should persist in this form or not.